ceph/rbd/initial_config.py

```diff
@@ -0,0 +1,562 @@
+                        data_pool:
+                        ec-pool-k-m: 2,1
+                        ec_profile:
+                        pg_num:
+                        pgp_num:
+                        ec_pg_num:
+                        ec_pgp_num:
+                        image_1:
+                            size:
+                            io_total:
+                        image_2:
+                            size:
+                            io_total:
+    """
+    log.debug(
+        f'config received for rbd_config: {kw.get("config", "Config not received, assuming default values")}'
+    )
+    pool_types = update_config(**kw)
+    config = kw.get("config", {})
+    ceph_version = int(config.get("rhbuild")[0])
+
+    if kw.get("client_node"):
+        client = get_node_by_id(ceph_cluster, kw.get("client_node"))
+    else:
+        client = ceph_cluster.get_nodes(role="client")[0]
+    rbd = Rbd(client)
+
+    for pool_type in pool_types:
+        pool_config = getdict(config.get(pool_type))
+        is_ec_pool = True if "ec" in pool_type else False
+        if config_rbd_multi_pool(
+            rbd=rbd,
+            multi_pool_config=pool_config,
+            is_ec_pool=is_ec_pool,
+            ceph_version=ceph_version,
+            config=config,
+            client=client,
+            is_secondary=kw.get("is_secondary"),
+        ):
+            log.error(f"RBD configuration failed for {pool_type}")
+            return None
+
+    return {"rbd": rbd, "client": client, "pool_types": pool_types}
+
+
+def initial_mirror_config(**kw):
+    """
+    Create replicated pools, ecpools or both and corresponding images based on arguments specified
+    Configure mirroring and enable required type of mirroring on the given pools and images
+    For all the images mentioned in the config, primary_cluster will be the primary cluster
+    and secondary_cluster will be the secondary cluster.
+
+    Note: If in any test case there is a requirement to create few images as primary from primary_cluster
+    and few other images as primary from secondary_cluster, then please call initial_mirror_config twice
+    by specifying only the required images to be primary on the given primary_cluster cluster
+    Args:
+        **kw:
+            "clusters":
+                "<cluster_1_name>":
+                    "config": <config_as_shown_in_example_below>"
+                "<cluster_2_name>":
+                    "config": <config_as_shown_in_example_below>"
+    Note: For configuration Advanced configuration <specifying only number of pools and images>,
+    num_pools number of pools will be created as specified in both cluster configurations,
+    num_images number of images will be created as primary in that pool, unless num_secondary_images
+    is exclusively specified, if it is specified then the rest of the images will be primary.
+
+    Examples: In default configuration, pool and image names will be taken as random values.
```

Comment:
since this example is too big can we `refer` this example for line optimization and for code readability?